### ISSUE TITLE
Spectral filters or spectral vanishing viscosity methods

### ISSUE THREAD

**Issue by glwagner** [glwagner/dedaLES#26]
We might consider implementing spectral filters, or a spectral vanishing viscosity method to remove energy from small-scales and (potentially) stabilize our solutions. This is a form of LES which has received some attention in the literature:

https://www.cscamm.umd.edu/people/faculty/tadmor/spectral_viscosity/Kar2_LES_SV_JCP163_2000.pdf

Please send more references if they are obtained.

Can we implement functions / methods that operate on equations in spectral space in dedalus?

**Comment by kburns** [comment]
Yes any pseudodifferential filter (multiplication in coefficient space) is pretty easy to implement.

**Comment by glwagner** [comment]
Hmm...

So there are a few methods. One I've used in the past is truly a 'filter', in that high wavenumbers are simply filtered out of the solution at the end of each time-step.

Another method is the 'spectral vanishing viscosity' (see the discussion in the above paper starting with eq 4), which is slightly different --- the viscous flux is formulated as a convolution of a 'viscosity kernel' with the gradient of the solution.

I'd like to implement the latter method. How do we implement convolution operators in dedalus?
